plugin/trino-hive/src/main/java/io/trino/plugin/hive/aws/athena/PartitionProjectionService.java

```diff
@@ -0,0 +1,315 @@
+                    rewriteProperty(
+                            columnMetadata.getProperties(),
+                            metastoreTableProperties,
+                            COLUMN_PROJECTION_TYPE,
+                            getMetastoreProjectionPropertyKey(columnMetadata.getName(), PROPERTY_SUFFIX_COLUMN_PROJECTION_TYPE),
+                            value -> value.toString().toLowerCase(Locale.ENGLISH));
+                    rewriteProperty(
+                            columnMetadata.getProperties(),
+                            metastoreTableProperties,
+                            COLUMN_PROJECTION_VALUES,
+                            getMetastoreProjectionPropertyKey(columnMetadata.getName(), PROPERTY_SUFFIX_COLUMN_PROJECTION_VALUES),
+                            value -> Joiner.on(",").join((List<String>) value));
+                    rewriteProperty(
+                            columnMetadata.getProperties(),
+                            metastoreTableProperties,
+                            COLUMN_PROJECTION_RANGE,
+                            getMetastoreProjectionPropertyKey(columnMetadata.getName(), PROPERTY_SUFFIX_COLUMN_PROJECTION_RANGE),
+                            value -> Joiner.on(",").join((List<String>) value));
+                    rewriteProperty(
+                            columnMetadata.getProperties(),
+                            metastoreTableProperties,
+                            COLUMN_PROJECTION_INTERVAL,
+                            getMetastoreProjectionPropertyKey(columnMetadata.getName(), PROPERTY_SUFFIX_COLUMN_PROJECTION_INTERVAL),
+                            String::valueOf);
+                    rewriteProperty(
+                            columnMetadata.getProperties(),
+                            metastoreTableProperties,
+                            COLUMN_PROJECTION_INTERVAL_UNIT,
+                            getMetastoreProjectionPropertyKey(columnMetadata.getName(), METASTORE_PROPERTY_KEY_SUFFIX_COLUMN_PROJECTION_INTERVAL_UNIT),
+                            value -> value.toString().toLowerCase(Locale.ENGLISH));
+                    rewriteProperty(
+                            columnMetadata.getProperties(),
+                            metastoreTableProperties,
+                            COLUMN_PROJECTION_DIGITS,
+                            getMetastoreProjectionPropertyKey(columnMetadata.getName(), PROPERTY_SUFFIX_COLUMN_PROJECTION_DIGITS),
+                            String::valueOf);
+                    rewriteProperty(
+                            columnMetadata.getProperties(),
+                            metastoreTableProperties,
+                            COLUMN_PROJECTION_FORMAT,
+                            getMetastoreProjectionPropertyKey(columnMetadata.getName(), PROPERTY_SUFFIX_COLUMN_PROJECTION_FORMAT),
+                            String::valueOf);
+                });
+
+        // If enabled we initialize PartitionProjection to validate properties
```

Comment:
Fair point.
My intention was to allow totally disable PP. 
This is because we might attach to partition projection properties created not created by Trino but by 3'd party system stored in HMS (pre-existing). 
That way we might get away with disabling Partition Projection and still be able to query this table.

But its right that we still need to overwrite at least enable property and may affect 3'd party system.

Maybe we should have a separate Trino only table property 🤔  to enable disable Partition Projection.
So what I propose is to:
- read Trino only enable property e.g. `trino.ignore.parition.projection`
- if above does not exist load HMS (Athena) enable property `projection.enabled` and as set trino property to it as well as `trino.ignore.parition.projection=!projection.enabled`
- store both but original Athena will remain unchanged. 

So our table might look in the end like so:
```
CREATE TABLE test (
  column1 varchar, 
  column2 varchar WITH (
    partition_projection_type='injected'
  )
) WITH (
  partition_projection_enabled=true
  trino_ignore_parition_projection=true
)
```

And from Trino PoV we will control if we want to bother with Partition Projection or not in case we just want to use table and there are some strange settings but we can't / shouldn't modify them.

Reply:
As of offline discussion we will add kill switch in config and session

Reply:
Actually, on the second look it looks that in  `getPartitionProjectionHiveTableProperties` I think we should validate always. As this method is called from flow when the table is created by Trino. So It does not make much sense to allow for broken values if those values are provided by user in `CREATE TABLE` statement.
Kill switch is needed for accessing tables with partition_projection_enabled, defined in Athena, in case Trino would not correctly parse specifics of table projection.